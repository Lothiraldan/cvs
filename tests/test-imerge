#!/bin/sh

echo "[extensions]" >> $HGRCPATH
echo "imerge=" >> $HGRCPATH
HGMERGE=true
export HGMERGE

hg init base
cd base

echo foo > foo
echo bar > bar
hg ci -Am0 -d '0 0'

hg mv foo foo2
echo foo >> foo2
hg ci -m1 -d '1 0'

hg up -C 0
echo bar >> foo
echo bar >> bar
hg ci -m2 -d '2 0'

echo % start imerge
hg imerge

cat foo2
cat bar

echo % status -v
hg -v imerge st

echo % next
hg imerge next

echo % merge next
hg --traceback imerge

echo % unresolve
hg imerge unres foo

echo % merge foo
hg imerge merge foo

echo % save
echo foo > foo2
hg imerge save ../savedmerge

echo % load
hg up -C 0
hg imerge --traceback load ../savedmerge
cat foo2

hg ci -m'merged' -d '3 0'
hg tip -v

echo % nothing to merge -- tip
hg imerge

hg up 0
echo % nothing to merge
hg imerge

exit 0
