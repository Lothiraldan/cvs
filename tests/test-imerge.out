adding bar
adding foo
1 files updated, 0 files merged, 1 files removed, 0 files unresolved
created new head
% start imerge
1 files updated, 0 files merged, 0 files removed, 0 files unresolved
(branch merge, don't forget to commit)
U foo
foo
bar
bar
bar
% status -v
merging e6da46716401 and 30d266f502e7
U foo (foo2)
% next
foo
% merge next
merging foo and foo2
all conflicts resolved
% unresolve
% merge foo
merging foo and foo2
all conflicts resolved
% save
% load
2 files updated, 0 files merged, 1 files removed, 0 files unresolved
2 files updated, 0 files merged, 0 files removed, 0 files unresolved
1 files updated, 0 files merged, 0 files removed, 0 files unresolved
(branch merge, don't forget to commit)
R foo
all conflicts resolved
foo
changeset:   3:fa9a6defdcaf
tag:         tip
parent:      2:e6da46716401
parent:      1:30d266f502e7
user:        test
date:        Thu Jan 01 00:00:03 1970 +0000
files:       foo foo2
description:
merged


% nothing to merge -- tip
abort: there is nothing to merge
2 files updated, 0 files merged, 1 files removed, 0 files unresolved
% nothing to merge
abort: there is nothing to merge - use "hg update" instead
% load unknown parent
abort: merge parent e6da46716401 not in repository
